src/api/app/controllers/source_controller.rb

```diff
@@ -93,7 +93,7 @@ def delete_package
     tpkg = Package.get_by_project_and_name(@target_project_name, @target_package_name,
                                            use_source: false, follow_project_links: false)
 
-    unless User.current.can_modify?(tpkg)
+    unless User.session!.can_modify?(tpkg)
```

Comment:
@coolo,

why you decided for ```session!``` here and not ```possibly_nobody```?

Reply:
as I wrote above - the action requires login. So in the end User.session would be the correct form, no need to double check. But I would prefer if this was cleaned up one by one - as @bgeuken and you said earlier, User.session! should be a smell, that indicates a move to pundit, especially in controllers.

Reply:
is it an exception for the reasoning 

"any access to policies and .can_* functions should prefer possibly_nobody to simplify the code"

because the action requires login and therefore we know that at this point of time, there will be a valid user other than nobody.. is that right, @coolo ?

Reply:
Yes, so possibly using User.session is the proper thing to do. But in the end all 3 functions return the same user if login is secured otherwise. So it's just a question of a good policy on what to do.